compiler/modules/bir/verify.bal

```diff
@@ -70,12 +70,60 @@ public function verifyFunctionCode(Module mod, FunctionDefn defn, FunctionCode c
     foreach BasicBlock b in code.blocks {
         check verifyBasicBlock(cx, b);
     }
+    check verifyGraph(cx, code.blocks, defn.position);
 }
 
 type IntBinaryInsn IntArithmeticBinaryInsn|IntBitwiseBinaryInsn;
 
 type Error err:Semantic|err:Internal;
 
+function verifyGraph(VerifyContext vc, BasicBlock[] blocks, Position pos) returns Error? {
+    boolean[] visited = from int b in 0 ..< blocks.length() select false;
+    check traveseGraph(vc, blocks, 0, visited, visited, pos);
+    if visited.indexOf(false) != () {
+        return vc.invalidErr("unreachable blocks in function code", pos);
+    }
+}
+
+function traveseGraph(VerifyContext vc, BasicBlock[] blocks, Label label, boolean[] visited, boolean[] currentGraph, Position pos) returns Error? {
+    if currentGraph[label] {
+        return vc.invalidErr("forward edge graph is not acyclic", pos);
+    }
+    if visited[label] {
+        return;
+    }
+    Label? onPanic = blocks[label].onPanic;
+    if onPanic is Label {
+        visited[onPanic] = true;
+    }
+    visited[label] = true;
+    currentGraph[label] = true;
+    Insn[] insns = blocks[label].insns;
+    var insnsLen = insns.length();
+    if insnsLen > 0 {
```

Comment:
Is having a block without any insn valid? I think not. We should make this a validation failure.